### PR TITLE
Fix aws_route53_record error when deserializing resource (null value is not allowed)

### PR DESCRIPTION
**Pull request by lotoussa**
| Q                 | A
| ----------------- | ---
| 🐛 Bug fix?       | yes
| 🚀 New feature?   | no
| ⚠ Deprecations?   | no
| ❌ BC Break       | no
| 🔗 Related issues | #375 
| ❓ Documentation  | no

## Description

Everything is in the #375 ticket.
We have also fixed new false positive.

### CODE REVIEW

**Review by eliecharra — APPROVED** [review]
LGTM but be careful to keep your PR minimalist.

The typo fix in `pkg/middlewares/route53_records_test.go` is not related to this bug

**Comment by eliecharra** [comment]
You need to rebase your PR onto branch `v0.7` to fix the linter issue in CI

**Review by sundowndev — APPROVED** [review]

File: pkg/resource/aws/aws_route53_record_ext.go [hunk 1]
```diff
@@ -27,6 +27,16 @@ func (r *AwsRoute53Record) NormalizeForState() (resource.Resource, error) {
 		r.SetIdentifier = aws.String("")
 	}
 
+	// This ensure that if we find a nil value we dont drift
```
Comment (wbeuil) [inline pkg/resource/aws/aws_route53_record_ext.go:30]: `This ensures that .... we don't drift`

File: pkg/resource/aws/aws_route53_record_ext.go [hunk 2]
```diff
@@ -27,6 +27,16 @@ func (r *AwsRoute53Record) NormalizeForState() (resource.Resource, error) {
...
+	if r.Records != nil && len(*r.Records) == 0 {
+		r.Records = nil
+	}
+
+	// This ensure that if we find a nil value we dont drift
```
Comment (wbeuil) [inline pkg/resource/aws/aws_route53_record_ext.go:35]: `This ensures that .... we don't drift`

File: pkg/resource/aws/aws_route53_record_ext.go [hunk 3]
```diff
@@ -46,6 +56,12 @@ func (r *AwsRoute53Record) NormalizeForState() (resource.Resource, error) {
 }
 
 func (r *AwsRoute53Record) NormalizeForProvider() (resource.Resource, error) {
+
+	// This ensure that if we find a nil value we dont drift
```
Comment (wbeuil) [inline pkg/resource/aws/aws_route53_record_ext.go:60]: `This ensures that .... we don't drift`

**Review by wbeuil — CHANGES_REQUESTED** [review]

File: pkg/resource/aws/aws_route53_record_ext.go (hunk 3 above)
Comment (lotoussa) [reply pkg/resource/aws/aws_route53_record_ext.go:60]: done, thank you

File: pkg/resource/aws/aws_route53_record_ext.go [hunk 4]
```diff
@@ -16,17 +16,27 @@ func (r *AwsRoute53Record) NormalizeForState() (resource.Resource, error) {
...
 	}
 
 	// On first run, this field is set to null in state file and to "" after one refresh or apply
-	// This ensure that if we find a nil value we dont drift
+	// This ensures that if we find a nil value we don't drift
 	if r.SetIdentifier == nil {
 		r.SetIdentifier = aws.String("")
 	}
 
+	// This ensures that if we find a nil value we don't drift
```
Comment (wbeuil) [inline pkg/resource/aws/aws_route53_record_ext.go:30]: `This ensures that if we find an empty records value we don't drift`

**Review by wbeuil — CHANGES_REQUESTED** [review]
One last modification

**Pull request merged**